### ISSUE TITLE
errorUtil code causes a TypeError when running in a Tauri app

### ISSUE THREAD

**Issue by bmwebster** [colinhacks/zod#3046]
Steps to replicate:

1. `import { z } from "zod";` into a web app running in [Tauri](https://github.com/tauri-apps/tauri)
2. Run application using `npm run tauri dev`

Expected: Application runs as usual.
Observed: Application fails with `TypeError: Cannot assign to read only property` logged to console.

This seems to be because Tauri by default freezes `Object.prototype` for security. [This Tauri issue](https://github.com/tauri-apps/tauri/issues/3416) describes the same issue occurring with different libraries.

There is a workaround of disabling the freezing of `Object.prototype`, but ideally it would be better to leave this enabled and still have the ability to use zod with Tauri.

Looking at the code that causes the issue, it seems to be [errorUtil.ts](https://github.com/colinhacks/zod/blob/master/src/helpers/errorUtil.ts#L5). This ends up as something like this in js:

```js
var errorUtil;
(function (errorUtil) {
    errorUtil.errToObj = (message) => typeof message === "string" ? { message } : message || {};
    errorUtil.toString = (message) => typeof message === "string" ? message : message === null || message === void 0 ? void 0 : message.message;
})(errorUtil = exports.errorUtil || (exports.errorUtil = {}));
```

Specifically it seems to be the `errorUtil.toString = ...` assignment that causes the error - renaming the const avoids the issue.

**Comment by lukeramsden** [comment]
I don't think this is stale but I don't know what causes it yet. Something else freezing a prototype somewhere clearly.

<img width="640" alt="Image" src="https://github.com/user-attachments/assets/760d664c-cb71-45d0-abf6-fe8e00d462a6" />

I've seen this in our Sentry. We're not using Tauri.
